src/windows/main_window.py

```diff
@@ -3061,6 +3065,10 @@ def __init__(self, *args):
         self.SelectionAdded.connect(self.addSelection)
         self.SelectionRemoved.connect(self.removeSelection)
 
+        # Connect toolbar float changes to movability/frozenness
```

Comment:
Imagining reading this without reading `freezeMainToolBar`, the function name implies that we freeze the tool bar whenever it gets docked/undocked. Including "Freeze toolbar IF the view is frozen" in the comment might clarify.

Reply:
Fair enough. I sort of take it as an _invitation_ to read `freezeMainToolBar` to learn the details (I mean, there's a lot that's implicit here, including how the `topLevelChanged` signal works), but point taken. How about...
```suggestion
        # Ensure toolbar is movable when floated (even with docks frozen)
```
That's not the _entirety_ of what `freezeMainToolBar()`  does (since it also freezes it when _re_-docked), but it's the main reason it exists.